tests/Module/MockFactory.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Tests\Module;
+
+use Packetery\Module\CurrencySwitcherFacade;
+use Packetery\Module\Framework\WpAdapter;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class MockFactory {
+	private TestCase $testCase;
+
+	public function __construct( TestCase $testCase ) {
+		$this->testCase = $testCase;
```

Comment:
I don't like it.

Reply:
The reason is in my next review comment few lines under https://github.com/Zasilkovna/WooCommerce/pull/589#discussion_r1724743523